cli/internal/cmd/upgradeapply.go

```diff
@@ -73,9 +75,19 @@ func runUpgradeApply(cmd *cobra.Command, _ []string) error {
 
 	imagefetcher := imagefetcher.New()
 	configFetcher := attestationconfigapi.NewFetcher()
-
 	applyCmd := upgradeApplyCmd{upgrader: upgrader, log: log, imageFetcher: imagefetcher, configFetcher: configFetcher}
-	return applyCmd.upgradeApply(cmd, fileHandler)
+	return applyCmd.upgradeApply(cmd, fileHandler, stableClientFactoryImpl)
+}
+
+type stableClientFactory func(kubeconfigPath string) (getConfigMapper, error)
+
+// needed because StableClient returns the bigger kubernetes.StableInterface.
+func stableClientFactoryImpl(kubeconfigPath string) (getConfigMapper, error) {
+	return kubernetes.NewStableClient(kubeconfigPath)
+}
+
+type getConfigMapper interface {
+	GetCurrentConfigMap(ctx context.Context, name string) (*corev1.ConfigMap, error)
```

Comment:
```suggestion
type configMapGetter interface {
	GetCurrentConfigMap(ctx context.Context, name string) (*corev1.ConfigMap, error)
```
As already mentioned [in another PR](https://github.com/edgelesssys/constellation/pull/2177#discussion_r1286740377), this should be a getter.

Reply:
it's also changed in the followup PR, so I let it be changed there 👍